File: 3105.py
class Solution:
    def longestMonotonicSubarray(self, nums: list[int]) -> int:
        res = 1
        curr = 1
        increasing = 0
        for i in range(1, len(nums)):
            if nums[i - 1] < nums[i]:
                if increasing > 0:
                    curr += 1
                else:
                    curr = 2
                    increasing = 1
            elif nums[i - 1] > nums[i]:
                if increasing < 0:
                    curr += 1
                else:
                    curr = 2 
                    increasing = -1
            else:
                increasing = 0
                curr = 1
            res = max(res, curr) 

        return max(res, curr)


nums = [1,4,3,3,2]
solution = Solution()
print(solution.longestMonotonicSubarray(nums))

File: solutions/3105.go
package main

import "fmt"


func longestMonotonicSubarray(nums []int) int {
	result := 0
	curr := 1
	increasing := 0
	for i := 0; i < len(nums) - 1; i++ {
		if nums[i] < nums[i + 1] {
			if increasing == 1 {
				curr++
			} else {
				increasing = 1
				curr = 2
			}
		} else if nums[i] > nums[i + 1] {
			if increasing == -1 {
				curr++
			} else {
				increasing = -1
				curr = 2
			}
		} else {
			increasing = 0
			curr = 1
		}
		result = max(result, curr)
	}
	return max(result, curr)
}

func main() {
	nums := []int{3,2,1}
	fmt.Println(longestMonotonicSubarray(nums))
}